Tools/clinic/clinic.py

```diff
@@ -1628,10 +1628,16 @@ def parse_clinic_block(self, dsl_name):
         def is_stop_line(line):
             # make sure to recognize stop line even if it
             # doesn't end with EOL (it could be the very end of the file)
-            if not line.startswith(stop_line):
-                return False
-            remainder = line[len(stop_line):]
-            return (not remainder) or remainder.isspace()
+            if line.startswith(stop_line):
+                remainder = line[len(stop_line):]
+                if remainder.isspace():
+                    return True
+                else:
+                    fail(f"Garbage after stop line: '{remainder.strip()}'")
```

Comment:
Maybe just show the raw string:

```suggestion
                    fail(f"Garbage after stop line: {remainder!r}")
```

Reply:
Yes, that's a nice improvement.